File: core/src/main/java/com/joshwiegand/abandon/managers/StateManager.java
package com.joshwiegand.abandon.managers;

import com.joshwiegand.abandon.states.GameOverState;
import com.joshwiegand.abandon.states.State;
import com.joshwiegand.abandon.states.HighScoreState;
import com.joshwiegand.abandon.states.MenuState;
import com.joshwiegand.abandon.states.PlayState;

public class StateManager {

  // current game state
  private State gameState;

  public static final int MENU = 'M' + 'E' + 'N' + 'U';
  public static final int PLAY = 'P' + 'L' + 'A' + 'Y';
  public static final int HIGH_SCORE = 'H' + 'I' + 'G' + 'H' + '_' + 'S' + 'C' + 'O' + 'R' + 'E';
  public static final int GAME_OVER ='G' + 'A' + 'M' + 'E' + '_' + 'O' + 'V' + 'E' + 'R';

  public StateManager() {
    setState(MENU);
  }

  public void setState(int state) {
    if (gameState != null) {
      gameState.dispose();
    }
    if (state == MENU) {
      gameState = new MenuState(this);
    }
    if (state == PLAY) {
      gameState = new PlayState(this);
    }
    if (state == HIGH_SCORE) {
      gameState = new HighScoreState(this);
    }
    if (state == GAME_OVER) {
      gameState = new GameOverState(this);
    }
  }

  public void update(float dt) {
    gameState.update(dt);
  }

  public void draw() {
    gameState.draw();
  }
}

File: old/abandon/core/src/main/kotlin/com/joshwiegand/abandon/managers/StateManager.kt
package com.joshwiegand.abandon.managers

import com.joshwiegand.abandon.states.GameOverState
import com.joshwiegand.abandon.states.State
import com.joshwiegand.abandon.states.HighScoreState
import com.joshwiegand.abandon.states.MenuState
import com.joshwiegand.abandon.states.PlayState

class StateManager {

    // current game state
    private var gameState: State? = null

    init {
        setState(MENU)
    }

    fun setState(state: Int) {
        if (this.gameState != null) {
            this.gameState!!.dispose()
        }
        if (state == MENU) {
            this.gameState = MenuState(this)
        }
        if (state == PLAY) {
            this.gameState = PlayState(this)
        }
        if (state == HIGH_SCORE) {
            this.gameState = HighScoreState(this)
        }
        if (state == GAME_OVER) {
            this.gameState = GameOverState(this)
        }
    }

    fun update(dt: Float) {
        this.gameState!!.update(dt)
    }

    fun draw() {
        this.gameState!!.draw()
    }

    companion object {

        val MENU = 'M'.toInt() + 'E'.toInt() + 'N'.toInt() + 'U'.toInt()
        val PLAY = 'P'.toInt() + 'L'.toInt() + 'A'.toInt() + 'Y'.toInt()
        val HIGH_SCORE = 'H'.toInt() + 'I'.toInt() + 'G'.toInt() + 'H'.toInt() + '_'.toInt() + 'S'.toInt() + 'C'.toInt() + 'O'.toInt() + 'R'.toInt() + 'E'.toInt()
        val GAME_OVER = 'G'.toInt() + 'A'.toInt() + 'M'.toInt() + 'E'.toInt() + '_'.toInt() + 'O'.toInt() + 'V'.toInt() + 'E'.toInt() + 'R'.toInt()
    }
}